microcosm_fastapi/audit.py

```diff
@@ -0,0 +1,311 @@
+
+        """
+        self.func = logging_info.function_name
+        self.operation = logging_info.operation_name
+
+    @property
+    def content_length(self):
+        content_length = self.request.headers.get("Content-Length")
+        if content_length is not None:
+            try:
+                return max(0, int(content_length))
+            except (ValueError, TypeError):
+                pass
+
+        return None
+
+    async def get_json(
+            self,
+    ) -> Optional[Any]:
+
+        data = None
+        try:
+            data = await self.request.json()
+        except JSONDecodeError:
+            pass
+        return data
+
+
+async def parse_response(response):
+    """
+    Parse a FastAPI response into a body, a status code, and headers
+
+    """
+    # Taken from SO: https://stackoverflow.com/questions/60778279/fastapi-middleware-peeking-into-responses
+    # Consuming FastAPI response and grabbing body here
+    resp_body = [section async for section in response.__dict__['body_iterator']]
+
+    # Repairing FastAPI response
+    response.__setattr__('body_iterator', AsyncIteratorWrapper(resp_body))
+
+    # Formatting response body for logging
+    try:
+        body = json.loads(resp_body[0].decode())
+    except:
+        body = str(resp_body)
+
+    return body, response.status_code, response.headers
+
+
+def create_audit_request(graph, options):
+    async def audit_request(graph, options: AuditOptions, request: Request, call_next):
+        """
+        Audit request
+
+        """
+        logger = getLogger(AUDIT_LOGGER_NAME)
+
+        request_context = graph.request_context(request)
+        request_info = RequestInfo(options, request, request_context, graph.metadata)
+
+        logging_info: LoggingInfo = graph.logging_data_map.get_entry(request.url.path, request.method)
+        request_info.set_operation_and_func_name(logging_info)
+
+        with elapsed_time(request_info.timing):
+            response = await call_next(request)
+
+        request_error = getattr(request.state, 'error', None)
+        if request_error is None:
+            await request_info.capture_response(response)
+        else:
+            request_info.capture_error(request_error)
+
+        if not should_skip_logging(request):
+            if request_info.status_code == 500:
+                # something actually went wrong; investigate
+                logger.warning(request_info.to_dict())
+
+            else:
+                # usually log at INFO; a raised exception can be an error or
+                # expected behavior (e.g. 404)
+                if not request_info.options.log_as_debug:
+                    logger.info(request_info.to_dict())
+                else:
+                    logger.debug(request_info.to_dict())
+
+        return response
+
+    return partial(audit_request, graph, options)
+
+
+@defaults(
+    include_request_body=typed(type=int, default_value=DEFAULT_INCLUDE_REQUEST_BODY),
```

Comment:
```suggestion
    include_request_body_status=typed(type=int, default_value=DEFAULT_INCLUDE_REQUEST_BODY),   
```

unless we really need compat with previous config I guess?

Reply:
I did a quick grep of our code base and couldn't see anyone really using this and I did a deeper search into ansible but it seems like it's no even used so yeah let's update this. Thanks for the suggestion.

Reply:
I've also updated the response_body one as well